StripeCore/StripeCore/Source/DownloadManager/DownloadManager.swift

```diff
@@ -53,177 +41,114 @@ import UIKit
             )
             configuration.urlCache = cache
             configuration.requestCachePolicy = .useProtocolCachePolicy
-            self.urlCache = cache
+            self.diskCache = cache
         }
 
         session = URLSession(configuration: configuration)
-
-        imageCache = [:]
-        pendingRequests = [:]
-        updateHandlers = [:]
-
-        imageCacheSemaphore = DispatchSemaphore(value: 1)
-        pendingRequestsSemaphore = DispatchSemaphore(value: 1)
-
+        self.analyticsClient = analyticsClient
         super.init()
     }
 }
 
 // MARK: - Download management
 extension DownloadManager {
-    public func downloadImage(url: URL, placeholder: UIImage?, updateHandler: UpdateImageHandler?) -> UIImage {
-        if updateHandler == nil {
-            return downloadImageBlocking(placeholder: placeholder, url: url)
-        } else {
-            return downloadImageAsync(url: url, placeholder: placeholder, updateHandler: updateHandler)
-        }
-    }
 
-    func downloadImageBlocking(placeholder: UIImage?, url: URL) -> UIImage {
+    /// Downloads an image from a provided URL, using either a synchronous method or an asynchronous method.
+    /// If no `updateHandler` is provided, this function will block the current thread until the image is downloaded. If an `updateHandler` is provided, the function does not wait for the download to finish and returns a placeholder image immediately instead. When the image finishes downloading, the `updateHandler` will be called with the downloaded image.
```

Comment:
Why not have separate async and sync methods?  Having a method be async/sync based on parameters feels surprising and complicated.

Reply:
Agree this is not optimal, but main goal of this refactor was to not modify the public interface and just refactor the internals. I think we should refactor usage of this to not need a blocking image download, but I'm proposing we keep the public contract of this class unmodified for this PR. Ideally we can just remove the sync usage later. I can log a JIRA if you agree!

Reply:
That's reasonable. This whole interface is very complicated. Like, besides async/sync, the return behavior is totally different based on the presence of `updateHandler` etc. It makes it really hard to tell if this PR keeps the old behavior. If you're confident in the correctness though, I'm okay to ship what we have and ticket up to fix this. I think breaking this out into multiple methods could make this a lot simpler.